src/Microsoft.Azure.WebJobs.Host/Bindings/ConverterManagerExtensions.cs

```diff
@@ -0,0 +1,234 @@
+using System.Reflection;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+    /// <summary>
+    /// Extensions for <see cref="IConverterManager"/> 
+    /// </summary>
+    public static class IConverterManagerExtensions
+    {
+        /// <summary>
+        /// Add a converter for the given Source to Destination conversion.
+        /// The typeConverter type is instantiated with the type arguments and constructorArgs is passed. 
+        /// </summary>
+        /// <typeparam name="TSource"></typeparam>
+        /// <typeparam name="TDestination"></typeparam>
+        /// <typeparam name="TAttribute"></typeparam>
+        /// <param name="converterManager"></param>
+        /// <param name="typeConverter"></param>
+        /// <param name="constructorArgs"></param>
+        public static void AddConverterBuilder<TSource, TDestination, TAttribute>(
+            this IConverterManager converterManager,
+            Type typeConverter, 
+            params object[] constructorArgs)
+            where TAttribute : Attribute
+        {
+            if (converterManager == null)
+            {
+                throw new ArgumentNullException("converterManager");
+            }
+            converterManager.AddConverterBuilder<TSource, TDestination, TAttribute>((typeSrc, typeDest) =>
+            {
+                var tuple = FindConverterMethod(typeConverter, constructorArgs, typeSrc, typeDest);
+                object instance = tuple.Item1;
+                MethodInfo method = tuple.Item2;
+
+                Func<object, object> converter = (input) =>
+                {
+                    return method.Invoke(instance, new object[] { input });
+                };
+
+                return converter;
+            });
+        }
+
+        /// <summary>
+        /// Add a converter for the given Source to Destination conversion.
+        /// </summary>
+        /// <typeparam name="TSource"></typeparam>
+        /// <typeparam name="TDestination"></typeparam>
+        /// <typeparam name="TAttribute"></typeparam>
+        /// <param name="converterManager"></param>
+        /// <param name="converterInstance"></param>
+        public static void AddConverterBuilder<TSource, TDestination, TAttribute>(
+          this IConverterManager converterManager,
+          object converterInstance)
+          where TAttribute : Attribute
+        {
+            if (converterManager == null)
+            {
+                throw new ArgumentNullException("converterManager");
+            }
+            converterManager.AddConverterBuilder<TSource, TDestination, TAttribute>((typeSrc, typeDest) =>
+            {
+                Func<object, object> converter = GetConverter(converterInstance, typeSrc, typeDest);
+                return converter;
+            });
+        }
+
+        // Find a Convert* method that is compatible with the given source and destiation types. 
+        // Signature will be like:
+        //    TIn Convert*(TOut) 
```

Comment:
We don't have a strong covariance story yet. Although the converters we're adding right now don't really demand that.  Table is aggressive in that we have rules:
```
ITableEntity  --> ITableEntity   : do nothing. Common for DynamicEntity, or C# types that derive from ITE. 
JObject--> ITableEntity : invoke poco serialization logic originally in Script 
T --> ITableEntity:  invoke poco  serialization logic in sdk. 
```

We solve it via ordering. Rules registered first get precedence.